adagrams/game.py

```diff
@@ -1,11 +1,106 @@
+import random
+
 def draw_letters():
-    pass
+    LETTER_POOL = {
+    "A": 9,
+    "B": 2,
+    "C": 2,
+    "D": 4,
+    "E": 12,
+    "F": 2,
+    "G": 3,
+    "H": 2,
+    "I": 9,
+    "J": 1,
+    "K": 1,
+    "L": 4,
+    "M": 2,
+    "N": 6,
+    "O": 8,
+    "P": 2,
+    "Q": 1,
+    "R": 6,
+    "S": 4,
+    "T": 6,
+    "U": 4,
+    "V": 2,
+    "W": 2,
+    "X": 1,
+    "Y": 2,
+    "Z": 1,
+}
+    hand = []
+    full_letter_list = []
+    for alpha, number in LETTER_POOL.items():
+        for i in range(number):
+            full_letter_list.append(alpha)
+    while len(hand) != 10:
+        letter = random.choice(full_letter_list)
+        hand.append(letter)
+        full_letter_list.remove(letter)    
+    return hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+    for letter in word:
+        if letter in letter_bank:
+            if word.count(letter) <=  letter_bank.count(letter):
```

Comment:
Here we've got a for loop, then nested within the for loop we have the `in` operator which also has linear time complexity, and then again, we have .count() which itself has linear time complexity. This ultimately leads us to O(n^3), or cubic, time complexity. 

Another approach for implementing this solution would be to use a dictionary.

You could create a dict that has letters in the `letter_bank` list as keys and the values would be how many times the letters appears. So if `letter_bank` was ["a", "a", "b"] then you could create dict like `avail_letters = {"a": 2, "b": 1} 

Then you could loop over every element in `word` and check if it is in the `avail_letters` dict. If the element in `word` was a key in `avail_letters` then you could decrement the value by 1. You'd use this logic to determine if `word` uses available letters from the list `letters`